testing-modules/junit-5-advanced/src/test/java/com/baeldung/displayname/ParameterizedTestNameGeneratorUnitTest.java

```diff
@@ -0,0 +1,37 @@
+package com.baeldung.displayname;
+
+import java.util.stream.Stream;
+
+import org.junit.jupiter.api.Named;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+class ParameterizedTestNameGeneratorUnitTest {
+
+    @ParameterizedTest
+    @MethodSource("argumentsProvider")
+    void testDefaultArgNamesForParameterizedTest(String givenArg) {
+        // Test
+    }
+
+    @ParameterizedTest(name = "Parameter with index {index} => {0}")
+    @MethodSource("argumentsProvider")
+    void testCustomArgNamesForParameterizedTest(String givenArg) {
+        // Test
+    }
+
+    @ParameterizedTest
+    @MethodSource("namedArguments")
+    void testCustomArgNamesForParameterizedTestUsingNamed(String givenArg) {
+        // Test
+    }
```

Comment:
Please bear in mind the test name convention givenXXX_whenYYY_thenZZZ.

Reply:
Done